modules/jsonld/src/main/scala/ch/epfl/bluebrain/nexus/rdf/jsonld/JsonLd.scala

```diff
@@ -0,0 +1,118 @@
+package ch.epfl.bluebrain.nexus.rdf.jsonld
+
+import cats.Monad
+import cats.data.EitherT
+import ch.epfl.bluebrain.nexus.rdf.Iri
+import ch.epfl.bluebrain.nexus.rdf.Iri.AbsoluteIri
+import io.circe.Json
+import io.circe.syntax._
+import cats.implicits._
+
+object JsonLd {
+
+  /**
+    * Resolve all IRIs inside @context.
+    *
+    * @param json     [[Json]] with context to resolve
+    * @param resolver context resolver
+    * @return [[Json]] with all the context IRIs resolved
+    */
+  def resolveContext[F[_]: Monad](
+      json: Json
+  )(resolver: AbsoluteIri => F[Option[Json]]): EitherT[F, ContextResolutionException, Json] = {
+
+    def inner(resolvedIds: List[AbsoluteIri], context: Json): EitherT[F, ContextResolutionException, Json] =
+      (context.asString, context.asArray, context.asObject) match {
+        case (Some(str), _, _) =>
+          val nextRef = Iri.absolute(str).toOption
+          // format: off
+          for {
+            next  <- EitherT.fromOption[F](nextRef, IllegalContextValue(str))
+            _     <- if (resolvedIds.contains(next)) EitherT.leftT[F, Unit](CircularContextDependency(next :: resolvedIds)) else EitherT.rightT[F, ContextResolutionException](())
+            res   <- resolveOrNotFound(next)
```

Comment:
The method `resolveOrNotFound` seems to be called just here. I think we can just replace the call by the actual code, since it is quite clear:
```
EitherT.fromOptionF(resolver(ref), ContextNotFound(ref))
```

But not a big deal if you prefer to keep it as is, this is just styling.